tests/security/grub_auth/grub_authorization.pm

```diff
@@ -0,0 +1,89 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: grub2 supports restricting access to boot menu entries when
+#          building their images/appliances,so that only specified
+#          users can boot selected menu entries.
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#81721, tc#1768659
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use base 'consoletest';
+
+my $sup_user     = 'admin';
+my $sup_passwd   = 'pw_admin';
+my $maint_user   = 'maintainer';
+my $maint_passwd = 'pw_maintainer';
+my $test_user    = 'tester';
+my $test_passwd  = 'pw_test';
+
+sub grub_auth_oper {
+    my $para = shift;
+    type_string("reboot\n");
+    if ($para eq "operator") {
+        assert_screen("grub_auth_boot_menu_entry");
+        send_key("ret");
+        assert_screen("grub_auth_super_user_login");
+        type_string("$sup_user\n");
+        type_string("$sup_passwd\n");
+    }
+    elsif ($para eq "maintainer") {
+        assert_screen("grub_auth_boot_menu_entry_maintainer");
+        send_key("down");
+        send_key("ret");
+        assert_screen("grub_auth_maintain_user_login");
+        type_string("$maint_user\n");
+        type_string("$maint_passwd\n");
+    }
+    elsif ($para eq "not_auth_user") {
```

Comment:
just a simple question, if the condition is correct user with error passwd , will it be left into this group , right?

Reply:
yes, let me add the code there.

Reply:
Added.